sbioapputils/app_runner/workflow_utils.py

```diff
@@ -0,0 +1,123 @@
+import argparse
+import os
+import warnings
+import yaml
+
+
+def _parse_workflow(request):
+    """Helper function to parse the workflow configuration."""
+    if request.get('workflow_name'):
+        workflow_loc = f"app/{request['workflow_name']}.yml"
+    else:
+        workflow_loc = "app/workflow.yml"
+        
+    with open(workflow_loc, "r") as stream:
+        try:
+            yaml_dict = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+            
+    name = yaml_dict['name']
+    stages = yaml_dict['stages']
+    parameters = yaml_dict['parameters']
+    return name, stages, parameters
+
+
+def dir_path(string):
+    """Function to check if the specified string is a directory."""
+    if os.path.isdir(string):
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+def validate_config(request, job_id):
```

Comment:
this function is very big and hard to read

Reply:
I've split it up now